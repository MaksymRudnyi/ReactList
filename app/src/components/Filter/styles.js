import styled from 'styled-components';

const Content = styled.div`
  display: flex;
  flex-direction: row;
`;
Content.displayName = 'Content';

const Label = styled.label`
	margin-right: 10px;
`;
Label.displayName = 'Label';

export {
	Content,
	Label,
};
