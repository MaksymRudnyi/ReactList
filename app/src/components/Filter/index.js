import React from 'react';
import PropTypes from 'prop-types';

import { Content, Label } from './styles';

const Filter = ({ specialities, onFilterChange }) => (
	<Content>
		{Object.keys(specialities).map((speciality, index) => (
			<Label key={speciality+index}>
				<input
					type="checkbox"
					name={speciality}
					value={speciality}
					onChange={onFilterChange}
				/>
				{speciality}
			</Label>
		))}
	</Content>
);

export default Filter;

Filter.propTypes = {
	onFilterChange: PropTypes.func.isRequired,
	specialities: PropTypes.objectOf(PropTypes.bool),
};


