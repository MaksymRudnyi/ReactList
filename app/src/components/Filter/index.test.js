import React from 'react';
import toJson from 'enzyme-to-json';

import { shallow } from 'enzyme';

import Filter from './';

describe('Item', () => {
	let sut;
	let props = {
		specialities: ['Excavations', 'Plumbing', 'Electrical'],
		onFilterChange: jest.fn(),
	};

	it('should match snapshot', () => {
		sut = shallow(<Filter {...props} />);

		expect(toJson(sut)).toMatchSnapshot();
	});
});
