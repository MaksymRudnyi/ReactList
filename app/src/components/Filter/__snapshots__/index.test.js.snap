// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item should match snapshot 1`] = `
<Content>
  <Label
    key="00"
  >
    <input
      name="0"
      onChange={[MockFunction]}
      type="checkbox"
      value="0"
    />
    0
  </Label>
  <Label
    key="11"
  >
    <input
      name="1"
      onChange={[MockFunction]}
      type="checkbox"
      value="1"
    />
    1
  </Label>
  <Label
    key="22"
  >
    <input
      name="2"
      onChange={[MockFunction]}
      type="checkbox"
      value="2"
    />
    2
  </Label>
</Content>
`;
